api/src/db/DynamoBusinessesDataClient.ts

```diff
@@ -0,0 +1,151 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+import { ExecuteStatementCommand, QueryCommand, QueryCommandInput } from "@aws-sdk/client-dynamodb";
+import { DeleteCommand, DynamoDBDocumentClient, GetCommand, PutCommand } from "@aws-sdk/lib-dynamodb";
+import { unmarshall } from "@aws-sdk/util-dynamodb";
+import { migrateData } from "@db/config/dynamoDbConfig";
+import { BusinessesDataClient } from "@domain/types";
+import { Business } from "@shared/userData";
+
+export const DynamoBusinessesDataClient = (
+  db: DynamoDBDocumentClient,
+  tableName: string
+): BusinessesDataClient => {
+  const doMigration = async (data: any): Promise<Business> => {
+    const migratedData = migrateData(data);
+    await put(migratedData);
+    return migratedData;
+  };
+
+  const findByIndex = (
+    indexName: string,
+    keyConditionExpression: string,
+    expressionAttributeValues: { [key: string]: any }
+  ): Promise<Business | undefined> => {
+    const params: QueryCommandInput = {
+      TableName: tableName,
+      IndexName: indexName,
+      KeyConditionExpression: keyConditionExpression,
+      ExpressionAttributeValues: expressionAttributeValues,
+    };
+
+    return db
+      .send(new QueryCommand(params))
+      .then((result) => {
+        if (!result.Items || result.Items.length !== 1) {
+          return;
+        }
+        const item = unmarshall(result.Items[0]);
+        return doMigration(item.data);
+      })
+      .catch((error) => {
+        console.log(error);
+        throw new Error("Not found");
+      });
+  };
+
+  const findByNAICSCode = (naicsCode: string): Promise<Business | undefined> => {
```

Comment:
[boulder] I'm not sure what the use cases for `findByNAICSCode` and `findByIndustry` are, especially given how their type definitions are written. Various businesses could have the same NAICS code and industry type, so it would get matched with multiple businesses if I'm understanding the code correctly, and wouldn't that cause line 36 to trigger a return? Please let know if I'm misunderstanding how it's supposed to work.

Reply:
Great question! You're right that multiple businesses can have the same NAICS code or industry type, and that’s actually the use case for these functions. Since we’ve set up secondary indexes on NAICS code and industry in the database, these functions are designed to query based on those indexes and return all matching businesses.
So, rather than triggering a return at line 36 after finding one match, the intention is to fetch and return all businesses that meet the criteria. I'll update the code

Reply:
Great point!  @somabadri You're absolutely right that multiple businesses can share the same NAICS code or industry. To address that, I added a findByIndexArray function, which is designed to return multiple businesses when querying by these indexes. This should prevent any unintended behavior on line 36 by ensuring that we're handling cases where more than one business matches the query. Let me know if you have any further thoughts or suggestions!